gramps/gen/filters/rules/_hassourceofbase.py

```diff
@@ -59,16 +59,12 @@ class HasSourceOfBase(Rule):
     description = "Matches objects who have a particular source"
 
     def prepare(self, db: Database, user):
-        if self.list[0] == "":
-            self.source_handle = None
-            self.nosource = True
-            return
-
-        self.nosource = False
-        try:
-            self.source_handle = db.get_source_from_gramps_id(self.list[0]).handle
-        except:
+        self.nosource = self.list[0] == ""
+        if self.nosource:
             self.source_handle = None
+        else:
+            source = db.get_source_from_gramps_id(self.list[0])
+            self.source_handle = source.handle if source else None
 
     def apply_to_one(self, db: Database, object: CitationBase) -> bool:
```

Comment:
@dsblank I'm not sure if this typing of `db` is correct
pylance is reporting [Variable not allowed in type expression](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportInvalidTypeForm)

Reply:
See PR #1950

Reply:
I don't know... it was working.